cvat-ui/src/reducers/models-reducer.ts

```diff
@@ -88,15 +89,29 @@ export default function (state = defaultState, action: ModelsActions | AuthActio
                 modelRunnerTask: null,
             };
         }
+        case ModelsActionTypes.GET_INFERENCES_SUCCESS: {
+            const { requestedInferenceIDs } = state;
+
+            return {
+                ...state,
+                requestedInferenceIDs: {
+                    ...requestedInferenceIDs,
+                    ...action.payload.requestedInferenceIDs,
+                },
+            };
+        }
         case ModelsActionTypes.GET_INFERENCE_STATUS_SUCCESS: {
-            const { inferences } = state;
+            const { inferences, requestedInferenceIDs } = state;
 
             if (action.payload.activeInference.status === 'finished') {
+                delete requestedInferenceIDs[action.payload.activeInference.id];
+
                 return {
                     ...state,
                     inferences: Object.fromEntries(
                         Object.entries(inferences).filter(([key]): boolean => +key !== action.payload.taskID),
                     ),
+                    requestedInferenceIDs: { ...requestedInferenceIDs },
```

Comment:
- Let's refactor a little bit with `lodash.omit` for both `inferences` and `requestedInferenceIDs`
- The same for `CANCEL_INFERENCE_SUCCESS`